packit_service/worker/handlers/abstract.py

```diff
@@ -161,12 +174,57 @@ class JobHandler(Handler):
     triggers: List[TheJobTriggerType]
 
     def __init__(
-        self, config: ServiceConfig, job_config: Optional[JobConfig], event: Event
+        self,
+        package_config: PackageConfig,
+        job_config: Optional[JobConfig],
+        event: dict,
     ):
-        super().__init__(config)
+        self.package_config: PackageConfig = package_config
         self.job_config: Optional[JobConfig] = job_config
+        self.event_type = (
+            EventType(event.get("event_type")) if event.get("event_type") else None
+        )
+        self.trigger = (
+            TheJobTriggerType(event.get("trigger")) if event.get("trigger") else None
+        )
+        self.user_login = event.get("user_login")
+        self.trigger_id = event.get("trigger_id")
+        self.project_url = event.get("project_url")
+
         self.event = event
+        self._db_trigger: Optional[AbstractTriggerDbType] = None
+        self._project: Optional[GitProject] = None
+        self._config: Optional[ServiceConfig] = None
         self._clean_workplace()
 
+    @property
+    def db_trigger(self):
+        if not self._db_trigger and self.trigger_id is not None:
+            if self.trigger in (
+                TheJobTriggerType.pull_request,
+                TheJobTriggerType.pr_comment,
+                TheJobTriggerType.pr_label,
+            ):
+                self._db_trigger = PullRequestModel.get_by_id(self.trigger_id)
+            elif self.trigger == TheJobTriggerType.issue_comment:
+                self._db_trigger = IssueModel.get_by_id(self.trigger_id)
+            elif self.trigger == TheJobTriggerType.release:
+                self._db_trigger = ProjectReleaseModel.get_by_id(self.trigger_id)
+            elif self.trigger in TheJobTriggerType.push:
+                self._db_trigger = GitBranchModel.get_by_id(self.trigger_id)
+        return self._db_trigger
+
+    @property
+    def project(self) -> Optional[GitProject]:
+        if not self._project:
+            self._project = self.get_project()
+        return self._project
+
+    def get_project(self) -> Optional[GitProject]:
```

Comment:
OK, I will merge them and and leave only `project`

Reply:
I see the point when using inheritance. I would only add a docstring, that describe the workflow:
- `get_project` will cause project instantiation each-time
- `project` property is lazzy and will run `get_project` when called for the first time

Reply:
> I see the point when using inheritance.

It does make sense in the Events, as subclasses can have their own `get_project` implementation, and the `project` property is implemented in the Event class and uses `get_project`, but since here is only one implementation for `get_project`, I would merge it as Hunor suggested.